src/database/import.ts

```diff
@@ -0,0 +1,114 @@
+import * as clc from "colorette";
+import pLimit from "p-limit";
+import { URL } from "url";
+import { Client } from "../apiv2";
+import { FirebaseError } from "../error";
+
+const MAX_CHUNK_SIZE = 1024 * 1024;
+const CONCURRENCY_LIMIT = 5;
+
+type Data = {
+  json: { [key: string]: any } | string | number | boolean;
+  pathname: string;
+};
+
+type ChunkedData = {
+  chunks: Data[] | null;
+  size: number;
+};
+
+/**
+ * Imports JSON data to a given RTDB instance.
+ *
+ * The data is parsed and chunked into subtrees of ~1 MB, to be subsequently written in parallel.
+ */
+export default class DatabaseImporter {
+  chunks: Data[];
+  private client: Client;
+  private limit = pLimit(CONCURRENCY_LIMIT);
+
+  constructor(private dbUrl: URL, file: string, private chunkSize = MAX_CHUNK_SIZE) {
+    let data;
+    try {
+      data = { json: JSON.parse(file), pathname: dbUrl.pathname };
```

Comment:
I'd like this command to support at least 100GB json files. So parsing everything to memory at once isn't really practical. https://github.com/FirebaseExtended/firebase-import uses JSONStream. Can we use it here as well?

Reply:
Done. Although, note that
1) `firebase-import` still loads the whole file into memory, not sure why, we'll not do that here
2) `JSONStream` assumes that the top-level objects can fit into memory but that the entire file should be streamed. I think this is a reasonable, maybe even necessary assumption.